### PR TITLE
UW-298 Add a forecast method that knows how to create a forecast namelist

### PR DESCRIPTION
**Pull request by venitahagerty**
**Description**

Jira issue UW-298
Github issue 244

**Type of change**

- [X ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [X ] pytests in GitHub actions.
- [X ] Tests on Mac laptop

**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by WeirAE — APPROVED** [review]
Nice & clean, LGTM

**Review by christinaholtNOAA — APPROVED** [review]
I agree! This looks good. I left one minor comment below.

File: tests/drivers/test_forecast.py [hunk 1]
```diff
@@ -4,14 +4,73 @@
...
+from uwtools import config
 from uwtools.drivers.driver import Driver
 from uwtools.drivers.forecast import FV3Forecast
 
+uwtools_file_base = os.path.join(os.path.dirname(__file__))
+
+
+def test_create_namelist():
+    """Tests create_namelist method with and without optional base file"""
+    forecast_obj = FV3Forecast(Driver)
```
[the remark this answers is not in the document]
Comment (venitahagerty) [reply tests/drivers/test_forecast.py:22]: FV3Forecast currently requires an argument, though now that I'm looking it doesn't seem to do anything with it.

[the remark this answers is not in the document]
Comment (WeirAE) [reply tests/drivers/test_forecast.py:22]: Could have been a syntax error on my part on initial setup, I thought I had to declare the ABC in the concrete class?

[the remark this answers is not in the document]
Comment (venitahagerty) [reply tests/drivers/test_forecast.py:22]: Driver is needed here to declare the subclass -
class FV3Forecast(Driver):

Should I remove argv from the __init__ in both classes?

**Review by christinaholtNOAA — APPROVED** [review]
LGTM!

File: tests/drivers/test_forecast.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (venitahagerty) [reply tests/drivers/test_forecast.py]: Removed Driver as an argument to FV3Forecast. Removed argv as a parameter to ``__init__``

**Comment by christinaholtNOAA** [comment]
@WeirAE Are you still okay with these changes?

**Review by WeirAE — APPROVED** [review]
Yep - if some strange circumstance causes us to add it back, we have several more steps to go and its still in the history, still approved!

**Pull request merged**